src/encord_active/lib/metrics/semantic/image_singularity.py

```diff
@@ -0,0 +1,115 @@
+from dataclasses import dataclass
+from typing import List
+
+import faiss
+import numpy as np
+from loguru import logger
+from sklearn.preprocessing import normalize as sk_normalize
+
+from encord_active.lib.common.iterator import Iterator
+from encord_active.lib.common.utils import (
+    fix_duplicate_image_orders_in_knn_graph_all_rows,
+)
+from encord_active.lib.embeddings.cnn import get_cnn_embeddings
+from encord_active.lib.embeddings.utils import LabelEmbedding
+from encord_active.lib.metrics.metric import DataType, EmbeddingType, Metric, MetricType
+from encord_active.lib.metrics.writer import CSVMetricWriter
+
+logger = logger.opt(colors=True)
+
+
+@dataclass
+class DataUnitInfo:
+    score: float
+    description: str
+
+
+class ImageSingularity(Metric):
+    def __init__(self, near_duplicate_threshold=0.97):
+        super(ImageSingularity, self).__init__(
+            title="Image Singularity",
+            short_description="Finds duplicate and near-duplicate images",
+            long_description=r"""
+This metric gives each image a score that shows each image's uniqueness.  
+- A score of zero means that the image has duplicates in the dataset; on the other hand, a score close to one represents that image is quite unique. Among the duplicate images, we only give a non-zero score to a single image, and the rest will have a score of zero (for example, if there are five identical images, only four will have a score of zero). This way, these duplicate samples can be easily tagged and removed from the project.    
+- Images that are near duplicates of each other will be shown side by side. 
+### Possible actions
+- **To delete duplicate images:** You can set the quality filter to cover only zero values (that ends up with all the duplicate images), then use bulk tagging (e.g., with a tag like `Duplicate`) to tag all images.
+- **To mark duplicate images:** Near duplicate images are shown side by side. Navigate through these images and mark whichever is of interest to you.
+""",
+            metric_type=MetricType.SEMANTIC,
+            data_type=DataType.IMAGE,
+            embedding_type=EmbeddingType.CLASSIFICATION,
+            annotation_type=None,
```

Comment:
If the suggestion above is applied this can be removed 

```suggestion
```
or explicitly set

```suggestion
            annotation_type=AnnotationType.NONE,
```